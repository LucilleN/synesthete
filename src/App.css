.App {
  /* margin-left: 5rem;
  margin-right: 5rem; */
  width: 100%;
  /* padding-left: 5rem;
  padding-right: 5rem; */
}
