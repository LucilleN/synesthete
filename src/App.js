import React, { useEffect } from 'react'
import { Route, BrowserRouter as Router, Switch } from 'react-router-dom'

import { MuiThemeProvider, createMuiTheme } from '@material-ui/core/styles'
import { withStyles } from '@material-ui/core/styles'

import Homepage from './pages/Homepage'
import About from './pages/About'
import Search from './pages/Search'
import UploadVisualization from './pages/UploadVisualization'
import Visualization from './pages/Visualization'
import MenuBar from './components/MenuBar'

import { apiHost } from './api'

export const theme = createMuiTheme({
  palette: {
    primary: {
      main: '#c23063', // magenta
      dark: '#180142' // dark purple
    },
    type: 'dark',
    dark: {
      pink: '#bf265c',
      purple: '#1c004f',
    },
    medium: {
      purple: '#5a0b66',
    },
    light: {
      pink: '#db91aa',
      purple: '#ac91db'
    },
    white: '#FFFFFF',
  },
  typography: {
    fontFamily: ['Dosis', 'Roboto', 'Open Sans', '"Helvetica Neue"', 'Arial', 'sans-serif'].join(',')
  }
})

export const defaultErrorText = 'Sorry, but something went wrong.'

const styles = () => ({
  root: {
    width: '100vw',
    height: '100%',
    minHeight: '100vh',
    background: `linear-gradient(0deg, ${theme.palette.dark.pink} 0%, ${theme.palette.dark.purple} 80%)`
  }
})

const routes = (
  <Router>
    <div>
      <MenuBar />
      <Switch>
        <Route exact path="/" component={Homepage} />
        <Route exact path="/about" component={About} />
        <Route exact path="/search" component={Search} />
        <Route exact path="/upload" component={UploadVisualization} />
        <Route exact path="/visualization" component={Visualization} />
        <Route exact path="/visualization/:songID" component={Visualization} />
      </Switch>
    </div>
  </Router>
)

const App = (props) => {

  useEffect(() => apiHost('https://api.spotify.com/v1/'))

  const { classes } = props

  return (
    <div className={classes.root}>
      <MuiThemeProvider theme={theme}>
        {routes}
      </MuiThemeProvider>
    </div>
  )
}

export default withStyles(styles)(App)
