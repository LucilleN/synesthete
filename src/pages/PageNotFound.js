import React from 'react'

import Typography from '@material-ui/core/Typography'
import { withStyles } from '@material-ui/core/styles'

import StartButton from '../components/StartButton'

const styles = theme => ({
  root: {
    textAlign: 'center',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
  },
  title: {
    color: theme.palette.light.pink,
    fontSize: '5rem',
    textAlign: 'center',
    marginTop: '30vh',
    opacity: 0.8,
  },
})

const PageNotFound = props => {
  const { classes } = props
  return (
    <div className={classes.root}>
      <Typography className={classes.title} id="title">
        page not found
      </Typography>
      <StartButton href="/" text="back to home" />
    </div>
  )
}

export default withStyles(styles)(PageNotFound)
