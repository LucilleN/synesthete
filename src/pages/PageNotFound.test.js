import React from 'react'
import ReactDOM from 'react-dom'
import TestRenderer from 'react-test-renderer'
import ReactTestUtils from 'react-dom/test-utils'

import { MuiThemeProvider } from '@material-ui/core/styles'

import { theme } from '../App'
import PageNotFound from './PageNotFound'

let component
beforeEach(() => {
  component = TestRenderer.create(
    <MuiThemeProvider theme={theme}>
      <PageNotFound />
    </MuiThemeProvider>
  )
})

describe('initial state', () => {
  let div
  beforeEach(() => {
    div = document.createElement('div')
    ReactTestUtils.act(() => {
      ReactDOM.render(
        <MuiThemeProvider theme={theme}>
          <PageNotFound />
        </MuiThemeProvider>
      , div)
    })
  })

  it('should have the title "page not found"', () => {
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    const pageTitle = div.querySelector('#title')
    expect(pageTitle.textContent).toBe('page not found')
  })

  it('should have a button that redirects to home', () => {
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    // const backToHome = div.querySelector('#back-to-home')
    // expect(backToHome).not.toBeNull()
  })
})

// describe('when the "back to home" button is clicked', () => {
//   let div
//   beforeEach(async() => {
//     div = document.createElement('div')
//     ReactTestUtils.act(() => {
//       ReactDOM.render(
//         <MuiThemeProvider theme={theme}>
//           <PageNotFound />
//         </MuiThemeProvider>
//       , div)
//     })
//     const button = div.querySelector('#back-to-home')
//     await ReactTestUtils.act(async () => {
//       await ReactTestUtils.Simulate.click(button)
//     })
//   })
//
//   it('should redirect from the page', () => {
//     const pageTitle = div.querySelector('#title')
//     expect(pageTitle).toBeNull()
//   })
// })
