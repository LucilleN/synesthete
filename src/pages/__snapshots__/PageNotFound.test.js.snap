// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initial state should have a button that redirects to home 1`] = `
<div
  className="PageNotFound-root-1"
>
  <p
    className="MuiTypography-root PageNotFound-title-2 MuiTypography-body1"
    id="title"
  >
    page not found
  </p>
  <a
    aria-disabled={false}
    aria-label="Start"
    className="MuiButtonBase-root MuiFab-root StartButton-fab-33 MuiFab-extended MuiFab-primary"
    href="/"
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
  >
    <span
      className="MuiFab-label"
    >
      <p
        className="MuiTypography-root StartButton-text-34 MuiTypography-body1"
      >
        back to home
      </p>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </a>
</div>
`;

exports[`initial state should have the title "page not found" 1`] = `
<div
  className="PageNotFound-root-1"
>
  <p
    className="MuiTypography-root PageNotFound-title-2 MuiTypography-body1"
    id="title"
  >
    page not found
  </p>
  <a
    aria-disabled={false}
    aria-label="Start"
    className="MuiButtonBase-root MuiFab-root StartButton-fab-33 MuiFab-extended MuiFab-primary"
    href="/"
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
  >
    <span
      className="MuiFab-label"
    >
      <p
        className="MuiTypography-root StartButton-text-34 MuiTypography-body1"
      >
        back to home
      </p>
    </span>
  </a>
</div>
`;
